src/nodes.coffee

```diff
@@ -885,30 +976,51 @@ exports.StatementLiteral = class StatementLiteral extends Literal
   compileNode: (o) ->
     [@makeCode "#{@tab}#{@value};"]
 
+  astType: ->
+    switch @value
+      when 'continue' then 'ContinueStatement'
+      when 'break'    then 'BreakStatement'
+      when 'debugger' then 'DebuggerStatement'
+
 exports.ThisLiteral = class ThisLiteral extends Literal
-  constructor: ->
+  constructor: (value) ->
     super 'this'
+    @shorthand = value is '@'
 
   compileNode: (o) ->
     code = if o.scope.method?.bound then o.scope.method.context else @value
     [@makeCode code]
 
+  astType: 'ThisExpression'
+  astProps: ['shorthand']
+
 exports.UndefinedLiteral = class UndefinedLiteral extends Literal
   constructor: ->
     super 'undefined'
 
   compileNode: (o) ->
     [@makeCode if o.level >= LEVEL_ACCESS then '(void 0)' else 'void 0']
 
+  astType: 'Identifier'
+  astProps:
+    name: 'value'
+
 exports.NullLiteral = class NullLiteral extends Literal
   constructor: ->
     super 'null'
 
+  astProps: []
+  emptyAst: yes
+
 exports.BooleanLiteral = class BooleanLiteral extends Literal
   constructor: (value, {@originalValue} = {}) ->
     super value
     @originalValue ?= @value
 
+  astProps: (o) ->
+    value: if @value is 'true' then yes else no
+    name: @originalValue
```

Comment:
Here's another Babel AST node type "extension": `name` gives us which version (eg `true`/`on`/`yes`) was used in the original source code so eg `name: 'yes'`

Reply:
@GeoffreyBooth I considered including a PR comment explaining the choice of `name` here. I was mimicking the `name` field on `Identifier` AST nodes, which seems quite similar (it's a string value corresponding to what in the source code is an unquoted "identifier"/keyword)

`rawValue` seems to have slightly different semantics, on both `NumericLiteral` and `StringLiteral` AST nodes, `rawValue` is the actual corresponding JS value of that node (eg `42` rather than `'42'` or `'abc'` rather than "'abc'"`)

But I'm glad we're both thinking along the lines of being inspired by existing Babel AST conventions when choosing how to extend/create new AST node types